### PR TITLE
New Module : aos_template as part of network/aos (clean)

### PR DESCRIPTION
**Pull request by dgarros**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
network/aos/aos_template

##### ANSIBLE VERSION
```
ansible 2.3.0 (aos_template_clean e06132a1fb) last updated 2017/02/09 14:56:31 (GMT -700)
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY
I'm working for [Apstra](http://www.apstra.com/), we are making a product to automate Datacenter Networks and we are developed a dozen of Ansible modules and one dynamic inventory to control our product with Ansible.
We have tried to follow Ansible's best practices as much as possible, all our modules are idempotent and support the mode `--check`.

This is the second batch of module that I'm submitting for review, the first `aos_ip_pool` #21044 has been recently approved

@gundalow, please can you take a look and let me know if I need to make some changes 

> I opened a similar PR yesterday for this module but it included the first PR #21044 so I closed it and opened a clean one #21173. Somehow git was not able to rebase properly  ...

### CODE REVIEW

**Review by jeremyschulman — APPROVED** [review]
shipit

File: lib/ansible/modules/network/aos/aos_template.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+  - Apstra AOS Template module let you manage your Template easily. You can create
+    create and delete Template by Name, ID or by using a JSON File. This module
+    is idempotent and support the I(check) mode. It's using the AOS REST API.
+requirements:
+  - "aos-pyez >= 0.6.0"
+options:
+  session:
+    description:
+      - An existing AOS session as obtained by aos_login module.
+    required: true
```
Comment (gundalow) [inline lib/ansible/modules/network/aos/aos_template.py:41]: You can remove the `required: false` lines, it's implicit

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by dgarros** [comment]
@gundalow I made the changes that you requested, thanks

**Review by gundalow — APPROVED** [review]

**Comment by dgarros** [comment]
ready_for_review

**Comment by gundalow** [comment]
shipit

**Pull request merged**